### PR TITLE
Fix parsing of traces with message handlers

### PR DESCRIPTION
**Pull request by nunomachado**
This PR fixes a bug that causes falcon-taz to crash when loading traces having message handlers at the end of the execution.

(follow-up on https://github.com/fntneves/falcon/pull/39)

### CODE REVIEW

**Comment by jcp19** [comment]
fix #43

File: falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ The message id is increased whenever a pair (snd,rcv) is complete */
     public Map<SocketEvent, List<Event>> handlerEvents;
 
     /* set indicating whether a given thread's trace has message handlers */
-    private HashSet hasHandlers;
+    public HashSet<String> hasHandlers;
```
Comment (fntneves) [inline falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: Why not keeping it private?

File: falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java [hunk 2]
```diff
@@ -537,27 +537,27 @@ private void parseMessageHandlers()
...
             SortedSet<Event> threadEvents = eventsPerThread.get( thread );
-            Iterator<Event> slowIt = threadEvents.iterator();
-            Iterator<Event> fastIt = threadEvents.iterator();
+            Iterator<Event> it = threadEvents.iterator();
 
-            while( slowIt.hasNext() )
-            {
-                Event e = slowIt.next();
-                fastIt.next(); // advance fastIt to follow slowIt
+            // we already know that this has at least one element
```
Comment (fntneves) [inline falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: Is it a requirement to make it work?

File: falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java [hunk 3]
```diff
@@ -537,27 +537,27 @@ private void parseMessageHandlers()
 
             // Use a fast and a slow iterator to handle nested message handlers while iterating through the thread events.
```
Comment (fntneves) [inline falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: This comment is now outdated.

File: falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java [hunk 4]
```diff
@@ -537,27 +537,27 @@ private void parseMessageHandlers()
...
             SortedSet<Event> threadEvents = eventsPerThread.get( thread );
-            Iterator<Event> slowIt = threadEvents.iterator();
-            Iterator<Event> fastIt = threadEvents.iterator();
+            Iterator<Event> it = threadEvents.iterator();
```
Comment (fntneves) [inline falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: Can we rename this to make it developer-friendly? :D

**Review by fntneves — CHANGES_REQUESTED** [review]

File: falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java (hunk 1 above)
Comment (jcp19) [reply falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: Fixed in commit a0b1e252fbe76fdb32b5a87bffef378b99dc2e79

File: falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java (hunk 3 above)
Comment (jcp19) [reply falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: Fixed in commit da28e4bde8df891167fb5590977c0fe30b8af086

File: falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java (hunk 4 above)
Comment (jcp19) [reply falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: Sure :+1: Is `threadIterator` ok?

Comment (jcp19) [reply falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java]: Fixed in 6903f5a53dc1a317b2dc8b638cb1036a89eba817

**Comment by fntneves** [comment]
Can it be closed without merging @nunomachado ?

**Pull request merged**